File: android/app/src/main/java/com/evan/chat/welcome/WelcomeActivity.java
package com.evan.chat.welcome;

import android.annotation.SuppressLint;
import android.content.pm.ActivityInfo;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.view.WindowManager;
import com.evan.chat.Injection;
import com.evan.chat.R;
import com.evan.chat.util.ActivityUtils;
import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.Fullscreen;
import org.androidannotations.annotations.UiThread;

/**
 * Created by IntelliJ IDEA
 * User: Evan
 * Date: 2018/2/19
 * Time: 下午1:23
 */
@SuppressLint("Registered")
@Fullscreen
@EActivity(R.layout.welcome_act)
public class WelcomeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        getWindow().addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION);

        WelcomeFragment welcomeFragment = (WelcomeFragment)getSupportFragmentManager().findFragmentById(R.id.contentFrame);

        if (welcomeFragment == null){
            welcomeFragment = WelcomeFragment.newInstance();
            ActivityUtils.addFragmentToActivity(getSupportFragmentManager(),welcomeFragment,R.id.contentFrame);
        }
        new WelcomePresenter(welcomeFragment,
                Injection.provideGetAutoUser(getApplicationContext()),
                Injection.provideUseCaseHandler(),
                Injection.provideSignInUser(getApplicationContext()));
        start();
    }

    @UiThread
    public void start(){
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
    }
}

File: android/app/src/main/java/com/evan/chat/welcome/WelcomeActivity.kt
package com.evan.chat.welcome

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.evan.chat.Injection
import com.evan.chat.R
import com.evan.chat.util.ActivityUtils

/**
 * Created by IntelliJ IDEA
 * User: Evan
 * Date: 2018/2/19
 * Time: 下午1:23
 */
@SuppressLint("Registered")
open class WelcomeActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.welcome_act)

//        window.addFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_NAVIGATION)

        var welcomeFragment: WelcomeFragment? = supportFragmentManager.findFragmentById(R.id.contentFrame) as? WelcomeFragment

        if (welcomeFragment == null) {
            welcomeFragment = WelcomeFragment.newInstance()
            ActivityUtils.addFragmentToActivity(supportFragmentManager, welcomeFragment, R.id.contentFrame)
        }
        WelcomePresenter(welcomeFragment,
                Injection.provideGetAutoUser(applicationContext),
                Injection.provideUseCaseHandler(),
                Injection.provideSignInUser(applicationContext),
                Injection.provideGetSetting(applicationContext),
                applicationContext)
    }
}
